sympy/assumptions/refine.py

```diff
@@ -26,6 +32,16 @@ def refine(expr, assumptions=True):
     >>> refine(sqrt(x**2), Q.positive(x))
     x
 
+    >>> refine(Q.real(x), Q.positive(x))
+    True
```

Comment:
I find this a bit strange. Why is `refine` being used rather than `ask` here?

In what situation does it make sense for `refine` to be applied to predicates like this?

Reply:
In `master`, `Relational` is evaluated to boolean by `refine`. While implementing `Q.eq`, I found out that other boolean expressions does not follow this so I made the design consistent.

For practical example, consider this situation.
```python
Or(Q.positive(x), Q.negative(x))
```
Reducing this expression using assumption `Q.nonnegative(x)` will return `Q.positive(x)`. We can't use `ask` here, because it will return `None`. In this situation, we need to use `refine`.